### ISSUE TITLE
Signatures 04: Use lists instead of Inner lists for §2.3 Prefixes

### ISSUE THREAD

**Issue by bblfish** [httpwg/http-extensions#1492]
The [Message Signing 04 $2.3 Prefixes](https://www.ietf.org/archive/id/draft-ietf-httpbis-message-signatures-04.html#name-list-prefixes) section gives inner lists as example of a data structure to which prefixes can be applied. 

```HTTP
X-List-A: (a b c d e f)
X-List-B: ()
```

The problem there is that this requires uniqueness of headers as [RFC 8941 Inner Lists](https://www.rfc-editor.org/rfc/rfc8941#name-inner-lists) on multiple headers fold into [RFC 8941 §3.1 Lists](https://www.rfc-editor.org/rfc/rfc8941#name-lists). So for example for a Message containing the following headers:

```HTTP
X-List-A: (a b c d e f)
X-List-A: (g h i)
```

would give the following folded header

```HTTP
X-List-A:  (a b c d e f), (g h i)
```

And that seems to be the more fundamental one on which Prefixes need to be applied.

**Comment by jricher** [comment]
The examples use the wrong syntax, probably an over-correction from when we switched the signature-input field to use an inner list in a dictionary. However, the other issues about folded header field values are still relevant.

See also #1315 for additional discussion of including these methods.

**Comment by richanna** [comment]
> The examples use the wrong syntax.... However, the other issues about folded header field values are still relevant.
> 
> See also #1315 for additional discussion of including these methods.

Can confirm this was just me getting the list syntax wrong. I'm not sure what "the other issues about folded header field values" are though?

**Comment by bblfish** [comment]
The fact that multiple headers of the same name (e.g. my `X-List-A` example above) do not fold into a single header of the same name with one inner list, made me wonder if the spec really intended to be speaking of inner lists in the first place. Given the RFC8941 also has Lists operated by commas, I decided to first ask here before implementing. So it looks like I was right to wait :-)

**Comment by jricher** [comment]
Fixed in #1529
